src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -61,6 +63,10 @@ public void onCompletion(RecordMetadata md, Exception e) {
 			LOGGER.error(e.getLocalizedMessage());
 			if ( e instanceof RecordTooLargeException ) {
 				LOGGER.error("Considering raising max.request.size broker-side.");
+				String kafkaFallbackTopic = this.context.getConfig().kafkaFallbackTopic;
+				if (kafkaFallbackTopic != null) {
+					publishFallbackTopic(kafkaFallbackTopic);
```

Comment:
We're doing `cc.markCompleted` below, which happens before the fallback message has been published, which may drop in-flight messages. What if we returned early here, and passed `cc` into the callback for the fallback publish? Or possibly just reuse this callback, as long as we don't get into a republish loop.

Reply:
same `cc` can be used, it'll delay clearing the inflight message list a bit.

Reply:
That's what we want (I think), but is there something we can do to prevent an infinite loop if somehow the truncated message is still too large?